test/TestUtils/project.json

```diff
@@ -0,0 +1,44 @@
+{
+  "version": "1.0.0-*",
+  "buildOptions": {
+    "compile": {
+      "include": [
+        "../../build_projects/shared-build-targets-utils/Utils/DotNetCli.cs",
+        "../../build_projects/shared-build-targets-utils/Utils/EnvVars.cs",
+        "../../build_projects/Microsoft.DotNet.Cli.Build.Framework/AnsiConsole.cs",
+        "../../build_projects/Microsoft.DotNet.Cli.Build.Framework/AnsiColorExtensions.cs",
+        "../../build_projects/Microsoft.DotNet.Cli.Build.Framework/ArgumentEscaper.cs",
+        "../../build_projects/Microsoft.DotNet.Cli.Build.Framework/BuildFailureException.cs",
+        "../../build_projects/Microsoft.DotNet.Cli.Build.Framework/BuildReporter.cs",
+        "../../build_projects/Microsoft.DotNet.Cli.Build.Framework/Command.cs",
+        "../../build_projects/Microsoft.DotNet.Cli.Build.Framework/CommandResult.cs",
+        "../../build_projects/Microsoft.DotNet.Cli.Build.Framework/Constants.cs",
+        "../../build_projects/Microsoft.DotNet.Cli.Build.Framework/Reporter.cs"
+      ]
+    }
+  },
+
+  "dependencies": {
+    "Microsoft.NETCore.App": {
+      "type": "platform",
+      "version": "1.0.0-rc3-004338"
```

Comment:
Just to clarify. You will only be able to run tests via the build. There are hacks required to make this work. This will create a circular dependency on a compatible sdk with the currently built host. So if the host makes a breaking change we will not get failing tests, the tests will fail to run. 

We will not be able to have a concept of a local dotnet-test loop, and the tests won't work in VS. This is because the targetted Microsoft.NETCore.App version will be higher than any version provided in the cli, because of the circular dependency. I honestly don't think these tradeoffs are worth it because it is going to slow down our dev cycle and these are some of the things I was hoping to get away from in the repo split.

@livarcocc @eerhardt If you think the tradeoffs are worth it I will go ahead with the fix though.

---
In reply to: [66113545](https://github.com/dotnet/core-setup/pull/97#discussion_r66113545) [](ancestors = 66113545,66108594,66107545)